service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteAzureContextFlightTest.java

```diff
@@ -0,0 +1,354 @@
+    UUID relayId = UUID.randomUUID();
+    ApiAzureRelayNamespaceCreationParameters creationParameters =
+        ControlledResourceFixtures.getAzureRelayNamespaceCreationParameters();
+    ControlledAzureRelayNamespaceResource relayNamespaceResource =
+        ControlledAzureRelayNamespaceResource.builder()
+            .common(
+                ControlledResourceFields.builder()
+                    .workspaceUuid(workspaceUuid)
+                    .resourceId(relayId)
+                    .name("wsm-test" + relayId.toString())
+                    .cloningInstructions(CloningInstructions.COPY_RESOURCE)
+                    .accessScope(AccessScopeType.fromApi(ApiAccessScope.SHARED_ACCESS))
+                    .managedBy(ManagedByType.fromApi(ApiManagedBy.USER))
+                    .build())
+            .namespaceName(creationParameters.getNamespaceName())
+            .region(creationParameters.getRegion())
+            .build();
+
+    controlledResourceService.createAzureRelayNamespace(
+        relayNamespaceResource, creationParameters, null, null, null, userRequest);
+    ControlledAzureRelayNamespaceResource gotRelayNamespaceResource =
+        controlledResourceService
+            .getControlledResource(workspaceUuid, relayId, userRequest)
+            .castByEnum(WsmResourceType.CONTROLLED_AZURE_RELAY_NAMESPACE);
+
+    assertEquals(relayNamespaceResource, gotRelayNamespaceResource);
+
+    // Delete the azure context.
+    FlightMap deleteParameters = new FlightMap();
+    deleteParameters.put(WorkspaceFlightMapKeys.WORKSPACE_ID, workspaceUuid.toString());
+    deleteParameters.put(JobMapKeys.AUTH_USER_INFO.getKeyName(), userRequest);
+
+    // Force each step to be retried once to ensure proper behavior.
+    Map<String, StepStatus> doFailures = new HashMap<>();
+    doFailures.put(
+        DeleteControlledAzureResourcesStep.class.getName(), StepStatus.STEP_RESULT_FAILURE_RETRY);
+    doFailures.put(DeleteAzureContextStep.class.getName(), StepStatus.STEP_RESULT_FAILURE_RETRY);
+    FlightDebugInfo debugInfo = FlightDebugInfo.newBuilder().doStepFailures(doFailures).build();
+
+    flightState =
+        StairwayTestUtils.blockUntilFlightCompletes(
+            jobService.getStairway(),
+            DeleteAzureContextFlight.class,
+            deleteParameters,
+            DELETION_FLIGHT_TIMEOUT,
+            debugInfo);
+    assertEquals(FlightStatus.SUCCESS, flightState.getFlightStatus());
+    assertTrue(testUtils.getAuthorizedAzureCloudContext(workspaceUuid, userRequest).isEmpty());
+    assertThrows(
+        ResourceNotFoundException.class,
+        () -> controlledResourceService.getControlledResource(workspaceUuid, relayId, userRequest));
+  }
+
+  // todo: this test would be better in the WorkspaceDeleteFlightTest, but that class extends
+  // BaseConnectedTest which doesn't have azure enabled... figure out what to do about this test
+  @Test
+  void deleteMcWorkspaceWithAzureContextAndResource() throws Exception {
```

Comment:
Looking for input on where to put this test. 

I've thought about moving it back to `WorkspaceDeleteFlightTest` and changing that test class to extend `BaseAzureTest` instead, but I'm worried that will change too much about when and where workspace deletion tests are run in our CI pipeline. 

Also thought about making a separate `AzureWorkspaceDeleteFlightTest`, but there isn't an actual `AzureWorkspaceDeleteFlight` so that's not great either. 

I suppose we could also just leave it here for now until `BaseConnectedTest` and `BaseAzureTest` converge, assuming that that will one day happen?

Reply:
Good question! I'll be interested in what other people think. My take is to leave it here for now.

We should have a larger conversation about how we want to structure WSM tests. Some of the factors I am aware of are the conflating of tags with the base classes; what tests should be integration tests vs connected tests; and what the needs are of the Verily deployment vs the Broad deployment. There are probably more things to think about.

Reply:
I think leaving it here is fine for now. It feels like a better fit for the `WorkspaceDeleteFlightTest` in the long run, where maybe we conditionally run it in the broad deployment, but leave it off in the verily?